### ISSUE TITLE
salt artifactory.downloaded module ignore classifier 

### ISSUE THREAD

**Issue by lisa2lisa** [saltstack/salt#23370]
salt --versions-report 
```
           Salt: 2015.2.0rc2
         Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
         Jinja2: 2.6
       M2Crypto: 0.21.1
 msgpack-python: 0.1.10
   msgpack-pure: Not Installed
       pycrypto: 2.6
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.6.0
           RAET: Not Installed
            ZMQ: 3.2.3
           Mako: 0.7.0
```
salt-minion --version
```
salt-minion 2015.2.0rc2 (Lithium)
```
state.sls file
```
xxx_downloaded:
  artifactory.downloaded:
    - artifact:
        artifactory_url: http://xxx/artifactory
        repository: 'libs-release-local'
        artifact_id: 'test'
        group_id: 'com.xxx.tools'
        packaging: 'jar'
        classifier: 'all'
        version: '0.1.0'
    - target_dir: /tmp/
```
basing on the state.sls file, the following url is the jar you want to download:
```
http://xxx/artifactory/libs-release-local/com/xxx/tools/test/0.1.0/test-0.1.0-all.jar
```
but the actual download file is:
```
http://xxx/artifactory/libs-release-local/com/xxx/tools/test/0.1.0/test-0.1.0.jar
```
the debug info is following
```
[DEBUG   ] release_url=http://xxx/artifactory/libs-release-local/com/xxx/tools/xxx/0.1.0/xxx-0.1.0.jar
```

**Comment by jfindlay** [comment]
This seems to be yet another unfortunate consequence of us only giving attention to documentation on the develop branch.  The `classifier` is available on develop but not 2015.2.  We are working on a better documentation system, but I am not directly involved with that effort, but it should be available in the next few months.

**Comment by rallytime** [comment]
If the `classifier` argument is only available on develop, then the docs should reflect that with a `versionadded` tag pointing to (in this case) Beryllium. I'm going to label this as a documentation fix.

**Comment by rallytime** [comment]
Oh, and the documentation effort that was mentioned above is being tracked in #12017.

**Comment by rallytime** [comment]
I have fixed this documentation discrepancy in #27639.
